src/main/io/osd.c

```diff
@@ -2993,6 +2991,19 @@ static void osdShowStats(void)
                 buff[4] = SYM_WH_KM_1;
                 buff[5] = '\0';
             }
+            // If traveled distance is less than 100 meters efficiency numbers are useless and unreliable so display --- instead
+            if (totalDistance < 10000) {
+                buff[0] = buff[1] = buff[2] = '-';
+                if (osdConfig()->stats_energy_unit == OSD_STATS_ENERGY_UNIT_MAH){
+                    buff[3] = SYM_MAH_KM_0;
+                    buff[4] = SYM_MAH_KM_1;
+                    buff[5] = '\0';
+                } else {
+                    buff[3] = SYM_WH_KM_0;
+                    buff[4] = SYM_WH_KM_1;
+                    buff[5] = '\0';
+                }
+            }
```

Comment:
This looks a bit unwieldy to me, but I can't think of a better way because the efficiency value does not have a fixed number of digits when using the `MAH` unit.